tests/index.test.tsx

```diff
@@ -1,14 +1,71 @@
 import React from "react";
-import { render } from "@testing-library/react";
-import { it } from "vitest";
+import "@vitest/web-worker";
+import { render, waitFor } from "@testing-library/react";
+import type { JsonProof } from "snarkyjs";
+import { describe, expect, it } from "vitest";
+import { createZKState } from "zk-states";
 
-// TODO: fix wasm error when importing zk-states library
-it("initializes state in a react component", async () => {
-  const Component = () => {
-    return <>hello world!</>;
-  };
+interface ZKState {
+  num: number;
+  incNum: () => void;
+}
 
-  const { findByText } = render(<Component />);
+const {
+  useInitZKStore,
+  useGetLatestProof,
+  useZKStore,
+  useProof,
+  useIsInitialized,
+} = createZKState<ZKState>(
+  new Worker(new URL("./worker.ts", import.meta.url), {
+    type: "module",
+  }),
+  (set) => ({
+    num: 0,
+    incNum: () => set((state) => ({ num: state.num + 1 })),
```

Comment:
Should we include some sort of assertion here to showcase the full usage of "actions"